Better error if there's nothing to deploy

**Is your feature request related to a problem? Please describe.**

this morning I wanted to re-deploy one of my stacks:

```
waypoint deploy -plain -app=FOO
! none available
zsh: exit 1  
```

hmm that doesnt make any sense. what does this mean?

```
WAYPOINT_LOG_LEVEL=debug waypoint deploy -plain -app=FOO
2021-01-13T14:06:24.780-0500 [INFO]  waypoint: waypoint version: full_string="Waypoint v0.2.0-110-g6f6b1fb5 (6f6b1fb5)" version=v0.2.0-110-g6f6b1fb5 prerelease= metadata= revision=6f6b1fb5
2021-01-13T14:06:24.781-0500 [DEBUG] waypoint: home configuration directory: path=/Users/orenmazor/Library/Preferences/waypoint
2021-01-13T14:06:24.781-0500 [INFO]  waypoint.server: attempting to source credentials and connect
2021-01-13T14:06:24.781-0500 [INFO]  waypoint.server: no server credentials found, using in-memory local server
2021-01-13T14:06:24.781-0500 [DEBUG] waypoint.server: opening local mode DB: path=data.db
2021-01-13T14:06:24.781-0500 [DEBUG] waypoint.server.singleprocess: checking if DB restore is requested
2021-01-13T14:06:24.782-0500 [DEBUG] waypoint.server.singleprocess: no restore file found, no DB restore requested
2021-01-13T14:06:24.838-0500 [INFO]  waypoint.server: starting built-in server for local operations: addr=127.0.0.1:59544
2021-01-13T14:06:24.838-0500 [INFO]  waypoint.server.http: HTTP listener not specified, HTTP API is disabled
2021-01-13T14:06:24.838-0500 [INFO]  waypoint.server.grpc: starting gRPC server: addr=127.0.0.1:59544
2021-01-13T14:06:24.839-0500 [INFO]  waypoint.server.grpc: /hashicorp.waypoint.Waypoint/SetServerConfig request
2021-01-13T14:06:24.892-0500 [INFO]  waypoint.server.grpc: /hashicorp.waypoint.Waypoint/SetServerConfig response: error=<nil> duration=52.53044ms
2021-01-13T14:06:24.893-0500 [INFO]  waypoint.server.grpc: /hashicorp.waypoint.Waypoint/GetVersionInfo request
2021-01-13T14:06:24.893-0500 [INFO]  waypoint.server.grpc: /hashicorp.waypoint.Waypoint/GetVersionInfo response: error=<nil> duration=147.283µs
2021-01-13T14:06:24.893-0500 [INFO]  waypoint: server version info: version=v0.2.0-110-g6f6b1fb5 api_min=1 api_current=1 entrypoint_min=1 entrypoint_current=1
2021-01-13T14:06:24.893-0500 [INFO]  waypoint: negotiated api version: version=1
2021-01-13T14:06:24.893-0500 [DEBUG] waypoint: will operate on app: name=FOO
2021-01-13T14:06:24.893-0500 [INFO]  waypoint.server.grpc: /hashicorp.waypoint.Waypoint/GetLatestPushedArtifact request
2021-01-13T14:06:24.893-0500 [INFO]  waypoint.server.grpc: /hashicorp.waypoint.Waypoint/GetLatestPushedArtifact response: error="rpc error: code = NotFound desc = none available" duration=61.855µs
! none available
2021-01-13T14:06:24.894-0500 [WARN]  waypoint: context cancelled, stopping interrupt listener loop
make: *** [schedule_tasks] Error 1
zsh: exit 2  
```

still doesn't make any sense. browsed the code: https://github.com/hashicorp/waypoint/blob/3695570c3915f9946319398f2c351addaaa380be/internal/server/singleprocess/state/app_operation.go#L308

that makes even less sense.

eventually after lunch I realized: I had to do a `docker system prune -af` this morning, so there is nothing to deploy. of cooooourse.

not a huge issue but definitely lost a bunch of momentum this morning.

**Describe the solution you'd like**
I get that changing the error message itself might not make sense. so maybe something earlier, or an additional debug message that would tell me it didnt actually find an artifact to deploy?

--- a/internal/server/singleprocess/state/app_operation.go
+++ b/internal/server/singleprocess/state/app_operation.go
@@ -323,7 +323,7 @@ func (op *appOperation) Latest(
 		}
 	}
 
-	return nil, status.Error(codes.NotFound, "none available")
+	return nil, status.Errorf(codes.NotFound, "No application named %q to deploy!", ref.Application)
 }
 
 // dbGet reads the value from the database.

--- a/internal/server/singleprocess/state/app_operation.go
+++ b/internal/server/singleprocess/state/app_operation.go
@@ -323,7 +323,7 @@ func (op *appOperation) Latest(
 		}
 	}
 
-	return nil, status.Errorf(codes.NotFound, "No application named %q to deploy!", ref.Application)
+	return nil, status.Errorf(codes.NotFound, "No application named %q is available!", ref.Application)
 }
 
 // dbGet reads the value from the database.

